devicetypes/ecolink/ecolink-dwzb1-eco-contact-sensor.src/ecolink-dwzb1-eco-contact-sensor.groovy

```diff
@@ -0,0 +1,243 @@
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+        configure()
+		result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+	}
+	return result
+}
+
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+
+	//check for tamper bit, send message to log if tampered/tamper cleared
+    if(zs.tamper != 0)
+    {
+    	sendEvent(name: "tamper", value: "detected")
+    }
+    else if(zs.tamper != 1)
+    {
+    	sendEvent(name: "tamper", value: "clear")
+    }
+    
+    // Some sensor models that use this DTH use alarm1 and some use alarm2 to signify motion
+	return (zs.isAlarm1Set() || zs.isAlarm2Set()) ? getContactResult('open') : getContactResult('closed')
+}
+
+private Map getBatteryResult(rawValue) {
+	log.debug "Battery rawValue = ${rawValue}"
+	def linkText = getLinkText(device)
+
+	def result = [:]
+
+	def volts = rawValue / 10
+
+	if (!(rawValue == 0 || rawValue == 255)) {
+		result.name = 'battery'
+		result.translatable = true
+		result.descriptionText = "{{ device.displayName }} battery was {{ value }}%"
+
+        def minVolts = 2.1
+        def maxVolts = 3.0
+        def pct = (volts - minVolts) / (maxVolts - minVolts)
+        def roundedPct = Math.round(pct * 100)
+        if (roundedPct <= 0)
+        roundedPct = 1
+        result.value = Math.min(100, roundedPct)
+        
+        //alert end user of low battery if batt < ~2.4V
+        if(result.value <= 25)
+        {
+        	result.descriptionText = "$device.displayName low battery alert!"
+        }
+	}
+    //if we get 0 its because the sensor didnt read the voltage prior to learning in. Assume battery is 100%
+    else if(rawValue == 0)
+    {
+    	result.name = 'battery'
+		result.translatable = true
+		result.descriptionText = "{{ device.displayName }} battery was {{ value }}%"
+        result.value = 100
+    }
+
+	return result
+}
+
+private Map getContactResult(value) {
+	log.debug 'contact'
+	String descriptionText = value == 'open' ? "{{ device.displayName }} contact opened" : "{{ device.displayName }} contact closed"
+	return [
+			name           : 'contact',
+			value          : value,
+			descriptionText: descriptionText,
+			translatable   : true
+	]
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    
+    log.debug "ping called"
+    
+    // Read the Battery Level and temperature
+    return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) + 
+    		zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000)
+            
+}
+
+def refresh() {
+	log.debug "refresh called"
+
+	//read battery voltage and temperature
+	def refreshCmds = zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+			zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000) +
+            zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, 0x0002)
+    
+    return refreshCmds
+}
+
+def configure() {
```

Comment:
`smartsense-open-closed-sensor.groovy` already has almost the same configuration as you've implemented. If binding with Poll Cluster is crucial for this model, you can define command for that more easily by utilizing `zigbee.addBinding(<add here Poll Cluster number>)`